pkg/utils/jolokia_client/jolokia_client.go

```diff
@@ -39,60 +41,72 @@ type JkInfo struct {
 	Ordinal string
 }
 
+// Get all matching broker pod infos for a give resource
+// parameters:
+// resource: the address CR for which the broker pods are gathered
+// ssInfos: target statefulsets to look for pods in
+// client: the client to access the api server
 func GetBrokers(resource types.NamespacedName, ssInfos []ss.StatefulSetInfo, client rtclient.Client) []*JkInfo {
 	reqLogger := ctrl.Log.WithValues("Request.Namespace", resource.Namespace, "Request.Name", resource.Name)
 
 	var artemisArray []*JkInfo = nil
 
-	for n, info := range ssInfos {
-		reqLogger.Info("Now retrieve ss", "order", n, "ssName", info.NamespacedName)
-		statefulset, err := ss.RetrieveStatefulSet(info.NamespacedName.Name, info.NamespacedName, info.Labels, client)
-		if nil != err {
-			reqLogger.Error(err, "error retriving ss")
-			reqLogger.Info("Statefulset: " + info.NamespacedName.Name + " not found")
-		} else {
-			reqLogger.Info("Statefulset: " + info.NamespacedName.Name + " found")
-			pod := &corev1.Pod{}
-			podNamespacedName := types.NamespacedName{
-				Name:      statefulset.Name + "-0",
-				Namespace: resource.Namespace,
+	for _, info := range ssInfos {
+		jkInfos := GetBrokersFromDNS(namer.SSToCr(info.NamespacedName.Name), info.NamespacedName.Namespace, info.Replicas, client)
+		artemisArray = append(artemisArray, jkInfos...)
+	}
+
+	reqLogger.Info("Gathered some mgmt array", "size", len(artemisArray))
+	return artemisArray
+}
+
+// Get brokers Using DNS names in the namespace
+func GetBrokersFromDNS(crName string, namespace string, size int32, client rtclient.Client) []*JkInfo {
+	reqLogger := ctrl.Log.WithValues("Request.Namespace", namespace, "Request.Name", crName)
+
+	var artemisArray []*JkInfo = nil
+	var i int32 = 0
+
+	clusterDomain := common.GetClusterDomain()
+	for i = 0; i < size; i++ {
+		// from NewHeadlessServiceForCR2 and
+		// https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#a-aaaa-records
+		ordinalFqdn := fmt.Sprintf("%s-ss-%d.%s-hdls-svc.%s.svc.%s", crName, i, crName, namespace, clusterDomain)
+
+		pod := &corev1.Pod{}
+		podNamespacedName := types.NamespacedName{
+			Name:      namer.CrToSS(crName) + "-" + strconv.Itoa(int(i)),
+			Namespace: namespace,
+		}
+
+		reqLogger.Info("Trying finding pod " + pod.Name)
```

Comment:
looks good,
I think reducing the verbosity of the logging will help, v(2) or something... really we only need to log when we have a problem.